### DISCUSSION TITLE
Does spatial computing work?

### DISCUSSION THREAD

**Discussion by mvvvv** [tesselode/kira#85] [Q&A]
I'm trying to implement kira for the stereokit binding I'm working on. I don't get any sound when I add the builder method `.output_destination(&emitter)`

        `if let Some(plane_sound) = open_asset(sk_info.clone(), "sounds/plane_engine.mp3") {
            let mut audio_manager = AudioManager::new(AudioManagerSettings::default()).unwrap();
            let mut spatial_scene = audio_manager.add_spatial_scene(SpatialSceneSettings::default()).unwrap();
            let emitter_settings = EmitterSettings::default();
            let emitter_settings =
                emitter_settings.distances(EmitterDistances::from((1.0, 1000.0))).persist_until_sounds_finish(true);
            let emitter = spatial_scene.add_emitter([5.0, 5.0, 5.0], emitter_settings).unwrap();
            let listener = spatial_scene
                .add_listener([0.0, 0.0, 0.0], Quat::IDENTITY.to_array(), ListenerSettings::default())
                .unwrap();
            let sound_settings = StaticSoundSettings::new().loop_region(0.0..35.0).output_destination(&emitter);
            audio_manager
                .play(StaticSoundData::from_media_source(plane_sound, sound_settings).unwrap())
                .unwrap();

            self.plane_sound_inst = Some(audio_manager);
        } `

**Comment by tesselode** [comment]
The code looks fine...can you post a whole project I can run?

**Reply by mvvvv** [reply]
Thank you for your reply. First The above example works fine on my ubuntu PC in its own main.rs. I should point out that StereoKit uses its own spatial audio by including miniaudio as you can see in this video :
https://youtu.be/s-16TH2gViQ

Unfortunatly miniaudio can't be easily disconnected from the project. The reason I'm testing Kira is to have spatial audio over greater distances.

I didn't publish the rust binding on crate.io so if you really  want to test it you'll have to download the project then the following template branch I created for you : https://github.com/mvvvv/stereokit-template/tree/kira-spatial-audio

**Comment by mvvvv** [comment]
I found the problem, it is tricky. The spatial_scene variable did not have sufficient life expectancy. Obviously you accept that it may have been dropped without saying anything which may not be very practical.
